### DISCUSSION TITLE
useCallback when working with uncontrolled forms

### DISCUSSION THREAD

**Discussion by enricoschaaf** [react-hook-form/react-hook-form#1788] [Q&A]
When I just work with really simple forms I don' t import react-hook-form und just use a normal uncontrolled form. I see a lot of examples wrapping the function that handles the submit in a useCallback, does this only have performance improvements when using controlled forms or should I also to that when working with uncontrolled forms?

I thought there is probably no one better to answer this question than the people there. 🙈

**Comment by bluebill1049** [comment]
i don't have much context, but you don't need to use `useCallback`. if you are building a simple form.

**Comment by enricoschaaf** [comment]
For example I saw this example in the Next.js docs and was wondering why useCallback() was used?
![carbon (1)](https://user-images.githubusercontent.com/54645197/83945813-763d6e00-a80d-11ea-829f-30cd38eeb818.png)
